lib/metadata/schema-parser.js

```diff
@@ -1098,7 +1098,7 @@ function isDoneForToken(replicationFactors, datacenters, replicasByDc) {
       continue;
     }
     var rf = Math.min(parseInt(replicationFactors[dcName], 10), dc.hostLength);
-    if (replicasByDc[dcName] < rf) {
+    if (rf != 0 && (!replicasByDc[dcName] || replicasByDc[dcName] < rf)) {
```

Comment:
`!=` operator should be avoided as it performs conversions. Its best to evaluate `rf !== 0` or `!rf`.